### DISCUSSION TITLE
Sorting many to many relation based on a field from a relation table

### DISCUSSION THREAD

**Discussion by sbkl** [prisma/prisma#10144] [Q&A] [answered]
Hi,

This is the way I'm currently sorting a model based on a field from a many to many relation table and wonder if there is a way to sort by calling directly the initial model? Thank you.

```ts
const relations = await prisma.categoriesOnPosts.findMany({
  where: { postId },
  orderBy: { sort: "asc" },
  include: { category: true },
});

return relations.map((r) => r.category);
```

**Comment by austincrim** [comment, marked as answer]
Hey @sbkl 👋,

Could you share the relevant models from your Prisma Schema so I can take a look at this? I think this should be possible, but I would need to accurately recreate your setup to confirm 👍.

**Reply by sbkl** [reply]
Hi @austincrim 
Sure. See below. I basically follow the docs example about many-to-many relationships. What I want to do is to be able to orderBy the sort field on the CategoriesOnPosts model when querying the Category model.

```prisma
model Post {
  id        String   @id @default(uuid())
  title     String
  body      String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  categories CategoriesOnPosts[]
}

model Category {
  id        String   @id @default(uuid())
  name      String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  posts CategoriesOnPosts[]
}

model CategoriesOnPosts {
  category   Category      @relation(fields: [categoryId], references: [id])
  categoryId String
  post       Post   @relation(fields: [postId], references: [id])
  postId     String
  sort       Int
  assignedAt DateTime  @default(now())

  @@id([categoryId, postId])
}
```

**Reply by austincrim** [reply]
You could try something like this:
```ts
const result = await prisma.category.findMany({
  include: { 
    posts: { orderBy: { sort: 'asc' } } 
  }
})
```
This will give you back the categories with the posts included and ordered by the `sort` field.

If this isn't quite the result you wanted, your original query looks good to me as well.

**Reply by sbkl** [reply]
That doesn't seem to work on a many to many relationship. But I'm good with the initial code as that's not many rows to query anyway.

Thank you for your answer!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
